engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java

```diff
@@ -2091,8 +2091,11 @@ public void execute(String[] args, ConsoleReader reader) throws Exception
           if (!commandLineInfo.localMode) {
 
             // see whether there is an app with the same name and user name running
-            String appNameAttributeName = StreamingApplication.DT_PREFIX + Context.DAGContext.APPLICATION_NAME.getName();
-            String appName = config.get(appNameAttributeName, appFactory.getName());
+            String appNameAttributeName = StreamingApplication.APEX_PREFIX + Context.DAGContext.APPLICATION_NAME.getName();
+            String appName = config.get(appNameAttributeName);
```

Comment:
It turns out that addDeprecation works both way, we can use old or new key to lookup old or new value. It does not matter which key is new or old. That wasn't how I read the documentation, but it works. Made the changes and added test for it.